### PR TITLE
fix(zboxapi): added SetWallet on mobile sdks

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed SetWallet issue for mobile sdks.

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0BoxSyncMac: https://github.com/0chain/0BoxSyncMac/pull/68

### CODE REVIEW

File: mobilesdk/zboxapi/client.go [hunk 1]
```diff
@@ -18,12 +18,29 @@ import (
...
+func SetWallet(clientID, clientPrivateKey, clientPublicKey string) error {
+	if zboxApiClient == nil {
+		return ErrZboxApiNotInitialized
+	}
+	if clientID != "" && clientPrivateKey != "" && clientPublicKey != "" {
+		zboxApiClient.SetWallet(clientID, clientPrivateKey, clientPublicKey)
+		return nil
+	}
+
+	return nil
```
Comment (peterlimg) [inline mobilesdk/zboxapi/client.go]: Shouldn't we return error if any of the client id or keys are empty?

File: mobilesdk/zboxapi/client.go [hunk 2]
```diff
@@ -18,12 +18,29 @@ import (
...
+	ErrZboxApiInvalidWallet  = errors.New("zboxapi: invalid wallet")
 )
 
 func Init(baseUrl, appType string) {
 	zboxApiClient = zboxapi.NewClient(baseUrl, appType)
-	zboxApiClient.SetWallet(client.GetClientID(), client.GetClientPrivateKey(), client.GetClientPublicKey())
+
+	c := client.GetClient()
+	if c != nil {
+		SetWallet(client.GetClientID(), client.GetClientPrivateKey(), client.GetClientPublicKey()) //nolint: errcheck
```
Comment (peterlimg) [inline mobilesdk/zboxapi/client.go]: Check error and log at least so that if err happens we could know what happened

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: mobilesdk/zboxapi/client.go (hunk 2 above)
Comment (cnlangzi) [reply mobilesdk/zboxapi/client.go]: fixed

File: mobilesdk/zboxapi/client.go (hunk 1 above)
Comment (cnlangzi) [reply mobilesdk/zboxapi/client.go]: fixed

**Review by peterlimg — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | fix/panicking_macos  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**